sql/timeseries.sql

```diff
@@ -496,3 +496,58 @@ CREATE OR REPLACE AGGREGATE last(value anyelement, rank anycompatible) (
     FINALFUNC = endpoint_final,
     FINALFUNC_EXTRA
 );
+
+-- When provided with a table, time source, data source,
+-- stride, and overall time range, this function will
+-- bin all data within the specified range into rows
+-- spaced according to the stride.
+--
+-- Time bins which lack data in the input table will still
+-- receive a single row of output with a NULL in the data
+-- column
+CREATE OR REPLACE FUNCTION @extschema@.date_bin_table(target_table_id regclass,
```

Comment:
I opened this PR to begin checking things in since I'm behind on these features. That said, I only this weekend figured out how to write a function that outputs arbitrary PostgreSQL record types through a kind of polymorphism abuse I didn't even know was possible.

The result of this is while this function here returns a series of (time, data) pairs, I am going to be changing it to automatically include all the columns of the target table, which will allow users to filter and group as they wish.